test/unit/test_ibm_primitives_v2.py

```diff
@@ -244,12 +246,44 @@ def test_no_session(self, primitive):
         service = backend.service
         inst = primitive(backend)
         inst.run(**get_primitive_inputs(inst))
-        self.assertIsNone(inst.session)
+        self.assertIsNone(inst.mode)
         service.run.assert_called_once()
         kwargs_list = service.run.call_args.kwargs
         self.assertNotIn("session_id", kwargs_list)
         self.assertNotIn("start_session", kwargs_list)
 
+    @data(SamplerV2, EstimatorV2)
+    def test_init_with_mode_param(self, primitive):
```

Comment:
Personally I'd split this into 3 test cases, one for each `mode` type, so it's easier to read. But I don't feel too strongly about it.

Reply:
Done in `4ee1b05`